File: app/src/main/java/com/androidhuman/example/simplecontacts/PeopleAdapter.java
package com.androidhuman.example.simplecontacts;

import com.androidhuman.example.simplecontacts.model.Person;

import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

class PeopleAdapter extends RecyclerView.Adapter<PeopleAdapter.PersonHolder> {

    private List<Person> people = null;

    @Override
    public PersonHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new PersonHolder(parent);
    }

    @Override
    public void onBindViewHolder(PersonHolder holder, int position) {
        Person person = people.get(position);

        holder.name.setText(person.getName());
        holder.address.setText(
                !TextUtils.isEmpty(person.getAddress())
                        ? person.getAddress() : "(No address)");
    }

    @Override
    public long getItemId(int position) {
        return people.get(position).getId();
    }

    @Override
    public int getItemCount() {
        return null != people ? people.size() : 0;
    }

    void setPeople(@Nullable List<Person> people) {
        this.people = people;
    }

    class PersonHolder extends RecyclerView.ViewHolder {

        TextView name;

        TextView address;

        PersonHolder(ViewGroup parent) {
            super(LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.item_person, parent, false));

            name = (TextView) itemView.findViewById(R.id.tv_item_person_name);
            address = (TextView) itemView.findViewById(R.id.tv_item_person_address);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    // TODO: Start an activity to edit an item.
                }
            });
        }
    }

}

File: app/src/main/kotlin/com/androidhuman/example/simplecontacts/PeopleAdapter.kt
package com.androidhuman.example.simplecontacts

import com.androidhuman.example.simplecontacts.model.Person
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.item_person.view.*

class PeopleAdapter : RecyclerView.Adapter<PeopleAdapter.PersonHolder>() {

    private var people: List<Person>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PersonHolder {
        return PersonHolder(parent)
    }

    override fun onBindViewHolder(holder: PersonHolder, position: Int) {
        val person = people!![position]
        //after
        with(holder.itemView){//with 는 kotlin에서 제공하는 함수.
            tv_item_person_name.text = person.name
            tv_item_person_address.text = if (!TextUtils.isEmpty(person.address))
                person.address
            else
                "(No address)"
        }
        //before
//        holder.name.text = person.name
//        holder.address.text = if (!TextUtils.isEmpty(person.address))
//            person.address
//        else
//            "(No address)"
    }

    override fun getItemId(position: Int): Long {
        return people!![position].id
    }

    override fun getItemCount(): Int {
        return if (null != people) people!!.size else 0
    }

    fun setPeople(people: List<Person>?) {
        this.people = people
    }

    inner class PersonHolder(parent: ViewGroup) : RecyclerView.ViewHolder(LayoutInflater.from(parent.context)
            .inflate(R.layout.item_person, parent, false))

}
